### ISSUE TITLE
Filenames with single quote and spaces raises SyntaxError in completer

### ISSUE THREAD

**Issue by gforsyth** [scopatz/xonsh#612]
I just ran into this one -- it's a bad idea to use `'` in your filenames, of course, but it can happen if people are really serious about possessives.

```
gil@bad_cat ~/test $ ls
gil's terrible filename  gil's.less.terrible.filename
```

trying to rename one of these and hitting Tab gives

```
gil@bad_cat ~/test $ mv gilException in thread Thread-198:
Traceback (most recent call last):
  File "/home/gil/miniconda3/lib/python3.4/threading.py", line r
    self.run()
  File "/home/gil/miniconda3/lib/python3.4/threading.py", line n
    self._target(*self._args, **self._kwargs)
  File "/home/gil/miniconda3/lib/python3.4/site-packages/promptn
    completions = list(buffer.completer.get_completions(documen)
  File "/home/gil/miniconda3/lib/python3.4/site-packages/xonsh/s
    self.ctx)
  File "/home/gil/miniconda3/lib/python3.4/site-packages/xonsh/e
    return self._filter_repeats(rtn), lprefix
  File "/home/gil/miniconda3/lib/python3.4/site-packages/xonsh/s
    canon = self._canonical_rep(comp)
  File "/home/gil/miniconda3/lib/python3.4/site-packages/xonsh/p
    x = ast.literal_eval(x)
  File "/home/gil/miniconda3/lib/python3.4/ast.py", line 46, inl
    node_or_string = parse(node_or_string, mode='eval')
  File "/home/gil/miniconda3/lib/python3.4/ast.py", line 35, ine
    return compile(source, filename, mode, PyCF_ONLY_AST)
  File "<unknown>", line 1
    'gil's terrible filename'
         ^
SyntaxError: invalid syntax
```

Only the first file causes trouble -- if I remove it, tab-completion behaves with the second file.

I just had a thought -- the file `gil's.less.terrible.filename` doesn't trigger quote-wrapping in the completer so there's no Syntax Error, but the spaces in `gil's terrible filename` cause the completer to wrap the whole filename.  Probably can fix this by checking for `'` in a completion target and then escaping it if quotes are needed.

**Comment by adqm** [comment]
Ah, yes, good catch!  We can be smarter about the quotes we use to wrap the string.

**Comment by gforsyth** [comment]
Fixed in #615
